File: backend/models/socials.go
package models

type Social struct {
	URL    string
	Usable bool
}

var socials = map[string]Social{
	"snapchat":  {URL: "https://snapchat.com/add/username", Usable: true},
	"youtube":   {URL: "https://youtube.com/@username", Usable: true},
	"discord":   {URL: "https://discord.com/users/id", Usable: true},
	"spotify":   {URL: "https://open.spotify.com/user/username", Usable: true},
	"instagram": {URL: "https://instagram.com/username", Usable: true},
	"tiktok":    {URL: "https://tiktok.com/@username", Usable: true},
	"telegram":  {URL: "https://t.me/username", Usable: true},
	"github":    {URL: "https://github.com/username", Usable: true},
	"roblox":    {URL: "https://www.roblox.com/users/id", Usable: true},
	"gitlab":    {URL: "https://gitlab.com/username", Usable: true},
	"twitch":    {URL: "https://twitch.tv/username", Usable: true},
	"namemc":    {URL: "https://namemc.com/profile/username", Usable: true},
	"steam":     {URL: "https://steamcommunity.com/id/username", Usable: true},
	"kick":      {URL: "https://kick.com/username", Usable: true},
	"x":         {URL: "https://x.com/username", Usable: true},
	"behance":   {URL: "https://behance.net/username", Usable: true},
	"litecoin":  {Usable: true},
	"bitcoin":   {Usable: true},
	"ethereum":  {Usable: true},
	"monero":    {Usable: true},
	"solana":    {Usable: true},
	"paypal":    {URL: "https://paypal.me/username", Usable: true},
	"reddit":    {URL: "https://reddit.com/user/username", Usable: true},
	"facebook":  {URL: "https://facebook.com/username", Usable: true},
	"ko-fi":     {URL: "https://ko-fi.com/username", Usable: true},
	"email":     {Usable: true},
	"pinterest": {URL: "https://pinterest.com/username", Usable: true},
	"custom":    {URL: "", Usable: true},
}

func GetSocials() map[string]Social {
	return socials
}

File: frontend/src/types/socials.ts
export interface SocialPlatformInfo {
  available: boolean;
  supportedTypes: {
    redirect: boolean;
    copy_text: boolean;
  };
  defaultType: 'redirect' | 'copy_text';
}

export const Socials: Record<string, SocialPlatformInfo> = {
  snapchat: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  youtube: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  discord: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  spotify: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  instagram: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  x: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  tiktok: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  telegram: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  github: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  roblox: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  gitlab: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  twitch: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  namemc: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  steam: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  kick: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  behance: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  litecoin: {
    available: true,
    supportedTypes: { redirect: false, copy_text: true },
    defaultType: 'copy_text'
  },
  bitcoin: {
    available: true,
    supportedTypes: { redirect: false, copy_text: true },
    defaultType: 'copy_text'
  },
  ethereum: {
    available: true,
    supportedTypes: { redirect: false, copy_text: true },
    defaultType: 'copy_text'
  },
  monero: {
    available: true,
    supportedTypes: { redirect: false, copy_text: true },
    defaultType: 'copy_text'
  },
  solana: {
    available: true,
    supportedTypes: { redirect: false, copy_text: true },
    defaultType: 'copy_text'
  },
  paypal: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  reddit: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  facebook: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  'ko-fi': {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  email: {
    available: true,
    supportedTypes: { redirect: false, copy_text: true },
    defaultType: 'copy_text'
  },
  pinterest: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
  custom: {
    available: true,
    supportedTypes: { redirect: true, copy_text: true },
    defaultType: 'redirect'
  },
};

export function isSocialTypeSupported(platform: string, type: 'redirect' | 'copy_text'): boolean {
  const platformInfo = Socials[platform.toLowerCase()];
  if (!platformInfo) return false;
  return platformInfo.supportedTypes[type];
}

export function getDefaultSocialType(platform: string): 'redirect' | 'copy_text' {
  const platformInfo = Socials[platform.toLowerCase()];
  if (!platformInfo) return 'redirect';
  return platformInfo.defaultType;
}